vulnerabilities/tests/test_openssl.py

```diff
@@ -20,466 +20,35 @@
-                        ),
-                    ),
-                    AffectedPackage(
-                        vulnerable_package=PackageURL(
-                            type="generic",
-                            namespace=None,
-                            name="openssl",
-                            version="1.1.1c",
-                        ),
-                        patched_package=PackageURL(
-                            type="generic",
-                            namespace=None,
-                            name="openssl",
-                            version="1.1.1e",
-                        ),
-                    ),
-                    AffectedPackage(
-                        vulnerable_package=PackageURL(
-                            type="generic",
-                            namespace=None,
-                            name="openssl",
-                            version="1.1.1d",
-                        ),
-                        patched_package=PackageURL(
-                            type="generic",
-                            namespace=None,
-                            name="openssl",
-                            version="1.1.1e",
-                        ),
-                    ),
-                ],
-                references=[
-                    Reference(
-                        reference_id="",
-                        url="https://github.com/openssl/openssl/commit/419102400a2811582a7a3d4a4e317d72e5ce0a8f",
-                        severities=[],
-                    ),
-                    Reference(
-                        reference_id="",
-                        url="https://github.com/openssl/openssl/commit/f1c5eea8a817075d31e43f5876993c6710238c98",
-                        severities=[],
-                    ),
-                ],
-            ),
-        ]
-
-        found_advisories = OpenSSLImporter.to_advisories(data)
-
-        found_advisories = list(map(Advisory.normalized, found_advisories))
-        expected_advisories = list(map(Advisory.normalized, expected_advisories))
-        assert sorted(found_advisories) == sorted(expected_advisories)
+TEST_DATA = os.path.join(BASE_DIR, "test_data", "openssl")
+
+
+class TestOpenssl(unittest.TestCase):
+    def test_to_advisory(self, regen=False):
+        expected_file = os.path.join(TEST_DATA, "openssl-expected.json")
+        result = [data.to_dict() for data in parse_test_advisory_data()]
+        if regen:
+            with open(expected_file, "w") as f:
+                json.dump(result, f, indent=2)
+            expected = result
+        else:
+            with open(expected_file) as f:
+                expected = json.load(f)
+        assert result == expected
+
+
+def parse_test_advisory_data() -> Iterable[AdvisoryData]:
+    xml_page = os.path.join(TEST_DATA, "openssl_xml_data.xml")
+    with open(xml_page) as f:
+        xml_response = f.read()
+    return parse_vulnerabilities(xml_response)
```

Comment:
Please test ``parse_vulnerabilites`` itself before using it in a test

Reply:
this test is designed to essentially test `parse_vulnerabilites` which is the core of OpenSSL Importer.

Reply:
Such a test should be called ``test_parse_vulnerabilites`` IMO

Reply:
IMHO the name of this test should be `test_openssl`
1. `OpensslImporter.advisory_data` making calls to 2 and 3
2. `fetch` returns XML string
3. `parse_vulnerabilities` parse XML string and makes call to 4
4. `to_advisory_data` parses individual issue

Where 2 is trivial and 1 is consuming the 3 as it's. So testing for 3 effectively becomes a robust test for OpenSSL importer as a whole.

Reply:
> IMHO the name of this test should be test_openssl

That would not be a **unit** test

Reply:
Yeah, that's true. Then we can name it `test_openssl_parse_vulnerabilites`. Does that sound good enough?

Reply:
@Hritik14 

> That would not be a unit test

That's OK though IMHO ... do you have a concern with this?

Reply:
@Hritik14 will add some more unit tests there, I guess that will justify the file name.